### ISSUE TITLE
Admin dashboard Most Viewed Products Tab issue (without product list)

### ISSUE THREAD

**Issue by rvpatel** [magento/magento2#1029]
Admin dashboard Most Viewed Products Tab issue click on not any product not display any message.
**Generate Error:**

![Screenshort](http://s2.postimg.org/ifcr3yb55/Mostviewproducts.png)

```
a:4:{i:0;s:339:"SQLSTATE[42S22]: Column not found: 1054 Unknown column 'e.entity_type_id' in 'on clause', query was: SELECT count(DISTINCT e.entity_id) FROM `report_event` AS `report_table_views`
 INNER JOIN `catalog_product_entity` AS `e` ON e.entity_id = report_table_views.object_id AND e.entity_type_id = 4 WHERE (report_table_views.event_type_id = 1)";i:1;s:8946:"#0 C:\xampp\htdocs\magento2\lib\internal\Magento\Framework\DB\Statement\Pdo\Mysql.php(95): Zend_Db_Statement_Pdo->_execute(Array)
#1 C:\xampp\htdocs\magento2\vendor\magento\zendframework1\library\Zend\Db\Statement.php(303): Magento\Framework\DB\Statement\Pdo\Mysql->_execute(Array)
#2 C:\xampp\htdocs\magento2\vendor\magento\zendframework1\library\Zend\Db\Adapter\Abstract.php(480): Zend_Db_Statement->execute(Array)
#3 C:\xampp\htdocs\magento2\vendor\magento\zendframework1\library\Zend\Db\Adapter\Pdo\Abstract.php(238): Zend_Db_Adapter_Abstract->query('SELECT count(DI...', Array)
#4 C:\xampp\htdocs\magento2\lib\internal\Magento\Framework\DB\Adapter\Pdo\Mysql.php(427): Zend_Db_Adapter_Pdo_Abstract->query('SELECT count(DI...', Array)
#5 C:\xampp\htdocs\magento2\vendor\magento\zendframework1\library\Zend\Db\Adapter\Abstract.php(828): Magento\Framework\DB\Adapter\Pdo\Mysql->query(Object(Magento\Framework\DB\Select), Array)
#6 C:\xampp\htdocs\magento2\lib\internal\Magento\Framework\Data\Collection\Db.php(205): Zend_Db_Adapter_Abstract->fetchOne(Object(Magento\Framework\DB\Select), Array)
#7 C:\xampp\htdocs\magento2\lib\internal\Magento\Framework\Data\Collection.php(260): Magento\Framework\Data\Collection\Db->getSize()
#8 C:\xampp\htdocs\magento2\lib\internal\Magento\Framework\Data\Collection.php(246): Magento\Framework\Data\Collection->getLastPageNumber()
#9 C:\xampp\htdocs\magento2\app\code\Magento\Eav\Model\Entity\Collection\AbstractCollection.php(1071): Magento\Framework\Data\Collection->getCurPage()
#10 C:\xampp\htdocs\magento2\app\code\Magento\Eav\Model\Entity\Collection\AbstractCollection.php(894): Magento\Eav\Model\Entity\Collection\AbstractCollection->_loadEntities(false, false)
#11 C:\xampp\htdocs\magento2\app\code\Magento\Backend\Block\Widget\Grid\Extended.php(464): Magento\Eav\Model\Entity\Collection\AbstractCollection->load()
#12 C:\xampp\htdocs\magento2\app\code\Magento\Backend\Block\Dashboard\Tab\Products\Viewed.php(70): Magento\Backend\Block\Widget\Grid\Extended->_prepareCollection()
#13 C:\xampp\htdocs\magento2\app\code\Magento\Backend\Block\Widget\Grid.php(396): Magento\Backend\Block\Dashboard\Tab\Products\Viewed->_prepareCollection()
#14 C:\xampp\htdocs\magento2\app\code\Magento\Backend\Block\Widget\Grid\Extended.php(491): Magento\Backend\Block\Widget\Grid->_prepareGrid()
#15 C:\xampp\htdocs\magento2\app\code\Magento\Backend\Block\Widget\Grid.php(457): Magento\Backend\Block\Widget\Grid\Extended->_prepareGrid()
#16 C:\xampp\htdocs\magento2\lib\internal\Magento\Framework\View\Element\AbstractBlock.php(636): Magento\Backend\Block\Widget\Grid->_beforeToHtml()
#17 C:\xampp\htdocs\magento2\app\code\Magento\Backend\Controller\Adminhtml\Dashboard\ProductsViewed.php(20): Magento\Framework\View\Element\AbstractBlock->toHtml()
#18 C:\xampp\htdocs\magento2\var\generation\Magento\Backend\Controller\Adminhtml\Dashboard\ProductsViewed\Interceptor.php(121): Magento\Backend\Controller\Adminhtml\Dashboard\ProductsViewed->execute()
#19 C:\xampp\htdocs\magento2\lib\internal\Magento\Framework\App\Action\Action.php(101): Magento\Backend\Controller\Adminhtml\Dashboard\ProductsViewed\Interceptor->execute()
#20 C:\xampp\htdocs\magento2\app\code\Magento\Backend\App\AbstractAction.php(222): Magento\Framework\App\Action\Action->dispatch(Object(Magento\Framework\App\Request\Http))
#21 [internal function]: Magento\Backend\App\AbstractAction->dispatch(Object(Magento\Framework\App\Request\Http))
#22 C:\xampp\htdocs\magento2\var\generation\Magento\Backend\Controller\Adminhtml\Dashboard\ProductsViewed\Interceptor.php(49): call_user_func_array(Array, Array)
#23 C:\xampp\htdocs\magento2\lib\internal\Magento\Framework\Interception\Chain\Chain.php(65): Magento\Backend\Controller\Adminhtml\Dashboard\ProductsViewed\Interceptor->___callParent('dispatch', Array)
#24 C:\xampp\htdocs\magento2\lib\internal\Magento\Framework\Interception\Chain\Chain.php(58): Magento\Framework\Interception\Chain\Chain->invokeNext('Magento\\Backend...', 'dispatch', Object(Magento\Backend\Controller\Adminhtml\Dashboard\ProductsViewed\Interceptor), Array, 'adminAuthentica...')
#25 C:\xampp\htdocs\magento2\app\code\Magento\Backend\App\Action\Plugin\Authentication.php(95): Magento\Framework\Interception\Chain\Chain->Magento\Framework\Interception\Chain\{closure}(Object(Magento\Framework\App\Request\Http))
#26 [internal function]: Magento\Backend\App\Action\Plugin\Authentication->aroundDispatch(Object(Magento\Backend\Controller\Adminhtml\Dashboard\ProductsViewed\Interceptor), Object(Closure), Object(Magento\Framework\App\Request\Http))
#27 C:\xampp\htdocs\magento2\lib\internal\Magento\Framework\Interception\Chain\Chain.php(63): call_user_func_array(Array, Array)
#28 C:\xampp\htdocs\magento2\lib\internal\Magento\Framework\Interception\Chain\Chain.php(58): Magento\Framework\Interception\Chain\Chain->invokeNext('Magento\\Backend...', 'dispatch', Object(Magento\Backend\Controller\Adminhtml\Dashboard\ProductsViewed\Interceptor), Array, 'designLoader')
#29 C:\xampp\htdocs\magento2\app\code\Magento\Core\App\Action\Plugin\Design.php(39): Magento\Framework\Interception\Chain\Chain->Magento\Framework\Interception\Chain\{closure}(Object(Magento\Framework\App\Request\Http))
#30 [internal function]: Magento\Core\App\Action\Plugin\Design->aroundDispatch(Object(Magento\Backend\Controller\Adminhtml\Dashboard\ProductsViewed\Interceptor), Object(Closure), Object(Magento\Framework\App\Request\Http))
#31 C:\xampp\htdocs\magento2\lib\internal\Magento\Framework\Interception\Chain\Chain.php(63): call_user_func_array(Array, Array)
#32 C:\xampp\htdocs\magento2\var\generation\Magento\Backend\Controller\Adminhtml\Dashboard\ProductsViewed\Interceptor.php(94): Magento\Framework\Interception\Chain\Chain->invokeNext('Magento\\Backend...', 'dispatch', Object(Magento\Backend\Controller\Adminhtml\Dashboard\ProductsViewed\Interceptor), Array, 'adminMassaction...')
#33 C:\xampp\htdocs\magento2\app\code\Magento\Backend\App\Action\Plugin\MassactionKey.php(33): Magento\Backend\Controller\Adminhtml\Dashboard\ProductsViewed\Interceptor->Magento\Backend\Controller\Adminhtml\Dashboard\ProductsViewed\{closure}(Object(Magento\Framework\App\Request\Http))
#34 [internal function]: Magento\Backend\App\Action\Plugin\MassactionKey->aroundDispatch(Object(Magento\Backend\Controller\Adminhtml\Dashboard\ProductsViewed\Interceptor), Object(Closure), Object(Magento\Framework\App\Request\Http))
#35 C:\xampp\htdocs\magento2\var\generation\Magento\Backend\Controller\Adminhtml\Dashboard\ProductsViewed\Interceptor.php(99): call_user_func_array(Array, Array)
#36 C:\xampp\htdocs\magento2\var\generation\Magento\Backend\Controller\Adminhtml\Dashboard\ProductsViewed\Interceptor.php(136): Magento\Backend\Controller\Adminhtml\Dashboard\ProductsViewed\Interceptor->___callPlugins('dispatch', Array, Array)
#37 C:\xampp\htdocs\magento2\lib\internal\Magento\Framework\App\FrontController.php(45): Magento\Backend\Controller\Adminhtml\Dashboard\ProductsViewed\Interceptor->dispatch(Object(Magento\Framework\App\Request\Http))
#38 [internal function]: Magento\Framework\App\FrontController->dispatch(Object(Magento\Framework\App\Request\Http))
#39 C:\xampp\htdocs\magento2\var\generation\Magento\Framework\App\FrontController\Interceptor.php(48): call_user_func_array(Array, Array)
#40 C:\xampp\htdocs\magento2\lib\internal\Magento\Framework\Interception\Chain\Chain.php(65): Magento\Framework\App\FrontController\Interceptor->___callParent('dispatch', Array)
#41 C:\xampp\htdocs\magento2\var\generation\Magento\Framework\App\FrontController\Interceptor.php(93): Magento\Framework\Interception\Chain\Chain->invokeNext('Magento\\Framewo...', 'dispatch', Object(Magento\Framework\App\FrontController\Interceptor), Array, 'install')
#42 C:\xampp\htdocs\magento2\lib\internal\Magento\Framework\Module\Plugin\DbStatusValidator.php(66): Magento\Framework\App\FrontController\Interceptor->Magento\Framework\App\FrontController\{closure}(Object(Magento\Framework\App\Request\Http))
#43 [internal function]: Magento\Framework\Module\Plugin\DbStatusValidator->aroundDispatch(Object(Magento\Framework\App\FrontController\Interceptor), Object(Closure), Object(Magento\Framework\App\Request\Http))
#44 C:\xampp\htdocs\magento2\var\generation\Magento\Framework\App\FrontController\Interceptor.php(98): call_user_func_array(Array, Array)
#45 C:\xampp\htdocs\magento2\var\generation\Magento\Framework\App\FrontController\Interceptor.php(122): Magento\Framework\App\FrontController\Interceptor->___callPlugins('dispatch', Array, Array)
#46 C:\xampp\htdocs\magento2\lib\internal\Magento\Framework\App\Http.php(115): Magento\Framework\App\FrontController\Interceptor->dispatch(Object(Magento\Framework\App\Request\Http))
#47 C:\xampp\htdocs\magento2\lib\internal\Magento\Framework\App\Bootstrap.php(245): Magento\Framework\App\Http->launch()
#48 C:\xampp\htdocs\magento2\index.php(39): Magento\Framework\App\Bootstrap->run(Object(Magento\Framework\App\Http))
#49 {main}";s:3:"url";s:116:"/magento2/admin/admin/dashboard/productsViewed/key/cfc3dd10eed272a4cf47917904c2a2b16dfadd8b46964c737804b3723c08b7f3/";s:11:"script_name";s:19:"/magento2/index.php";}

```

**Comment by kokoc** [comment]
@rvpatel Thank you for reporting the issue. 
Internal reference - MAGETWO-33894

**Comment by duongquyet** [comment]
did you fix that?

**Comment by vpelipenko** [comment]
@rvpatel, this issue is assigned to internal team and now it is "in progress" state. Please, stay tuned. We'll inform you when it is delivered to public repository.

**Comment by snohe** [comment]
@rvpatel, this issue has now been fixed in 0.74.0-beta1.  Thank you again for submitting this issue.  Feel free to reach out to us if you have any other questions.  Thank you for using Magento!
